gammapy/modeling/parameter.py

```diff
@@ -14,6 +16,19 @@
 log = logging.getLogger(__name__)
 
 
+class ParameterTypes(Enum):
```

Comment:
Assigning the model categories to parameters seems a bad idea to me. It adds complexity but what does it solve?

Reply:
It seemed the more general solution to me. Possibly we get more parameter types in future (could imagine e.g., distinguishing position from morphology parameters). And already now a `SpatialModel` could in fact declare "spectral" parameters e.g. to handle for energy dependent morphology, such as a spatially smoothly varying spectral index. But I see your point, given the current convention we have!

I guess the alternative and even simpler solution here is to just work with the `.is_norm` flag for now? And declare this on init of the parameter?

Reply:
I guess it mostly depends on the definition we employ for the `Parameter.type` attribute. One could even introduce `Parameter.model_type` and distinguish both cases...